### PR TITLE
[14.0][FWP] account_multicurrency_revaluation

### PR DESCRIPTION
**Pull request by MiquelRForgeFlow**
Forward port of https://github.com/OCA/account-closing/pull/200.

### CODE REVIEW

File: account_multicurrency_revaluation/model/account.py [hunk 1]
```diff
@@ -13,12 +15,72 @@ class AccountAccountLine(models.Model):
...
+
+    def _compute_revaluation_origin_line_count(self):
+        for line in self:
+            line.revaluation_origin_line_count = len(line.revaluation_origin_line_ids)
+
+    def action_view_revaluation_origin_lines(self):
+        self.ensure_one()
+        action = self.env["ir.actions.act_window"].for_xml_id(
+            "account", "action_account_moves_all"
+        )
```
Comment (GuillemCForgeFlow) [inline account_multicurrency_revaluation/model/account.py]:
```suggestion
        action = self.env["ir.actions.act_window"]._for_xml_id(
            "account.action_account_moves_all"
        )
```

File: account_multicurrency_revaluation/model/account.py [hunk 2]
```diff
@@ -13,12 +15,72 @@ class AccountAccountLine(models.Model):
...
+            action["res_id"] = self.revaluation_origin_line_ids.id
+        else:
+            action = {"type": "ir.actions.act_window_close"}
+        return action
+
+    def action_view_revaluation_created_line(self):
+        self.ensure_one()
+        action = self.env["ir.actions.act_window"].for_xml_id(
+            "account", "action_account_moves_all"
+        )
```
Comment (GuillemCForgeFlow) [inline account_multicurrency_revaluation/model/account.py]:
```suggestion
        action = self.env["ir.actions.act_window"]._for_xml_id(
            "account.action_account_moves_all"
        )
```
Same here.

File: account_multicurrency_revaluation/model/account.py [hunk 3]
```diff
@@ -82,10 +167,12 @@ def _revaluation_query(self, revaluation_date):
         aml.currency_id,
         aml.debit,
         aml.credit,
-        aml.amount_currency
+        aml.amount_currency,
+        aml.id as origin_aml_id
     FROM """
             + tables
             + """
+    LEFT JOIN account_move am ON aml.move_id = am.id
```
Comment (GuillemCForgeFlow) [inline account_multicurrency_revaluation/model/account.py:174]:
```suggestion
```
It is now added by the tables argument.

Comment (GuillemCForgeFlow) [reply account_multicurrency_revaluation/model/account.py:174]: Nevermind, before this commit https://github.com/odoo/odoo/commit/c2a72c268e6a28d7448b6dbf96fed4e36d24b3fe Odoo was using the move_id.state which caused the method _where_calc to add a left join with the account_move table. After that, the left join is not added anymore.

So we need to add it manually.

**Review by GuillemCForgeFlow — APPROVED** [review]
LGTM 👍🏽

**Review by AaronHForgeFlow — APPROVED** [review]
Code LG!

**Pull request merged**